portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactsService.java

```diff
@@ -152,22 +168,25 @@ private static StructuredName convertSinglePersonName(Name personName) {
 
   @Override
   public ContactsModelWrapper export(ExportInformation continuationInformation) throws IOException {
-    // TODO(olsona): implement pagination
-
-    Collection<VCard> vCards = new HashSet<>();
-
-    List<String> resourceNames = peopleService.contactGroups().batchGet().getResourceNames();
-    for (String resourceName : resourceNames) {
-      Person person = peopleService.people().get(resourceName).execute();
-      VCard vcard = convertToModel(person);
-      vCards.add(vcard);
+    // TODO(olsona): get next page using pagination token
+    Connections.List connectionsListRequest = peopleService.people().connections()
+        .list(SELF_RESOURCE);
+    ListConnectionsResponse response = connectionsListRequest.execute();
+    List<Person> initialPeopleList = response.getConnections();
+    List<VCard> vCards = new LinkedList<>();
+    for (Person initialPerson : initialPeopleList) {
```

Comment:
if you want to get fancy (probably a separate PR), you could do all these calls in parallel with Futures, so we're not blocking on each request

or better yet, a batch request if they have it..

Reply:
I believe they do have a batch request function, I'll use it in the next PR.